crates/json-abi/src/item.rs

```diff
@@ -9,19 +9,17 @@ use serde::{Deserialize, Deserializer, Serialize, Serializer};
 macro_rules! abi_items {
     ($(
         $(#[$attr:meta])*
-        $vis:vis struct $name:ident : $name_lower:literal {$(
+        $vis:vis struct $name:ident : $($name_lower:tt),+ {$(
             $(#[$fattr:meta])*
             $fvis:vis $field:ident : $type:ty,
         )*}
     )*) => {
+        trait FlattenStateMutability {
```

Comment:
I think we can simplify this and flatten at serde-time in `StateMutability`. There's also `constant: bool` alongside `payable: bool`

Reply:
does `constant: bool` effect the generation of the bindings at all? I totally agree with flattening it but do you think we can just ignore the `constant` field?

Reply:
I assume it's converted like this
```rust
match (constant, payable) {
    (false, false) => StateMutability::NonPayable,
    (false, true) => StateMutability::Payable,
    (true, false) => StateMutability::View,
    (true, true) => StateMutability::Payable, // I don't think this can happen
}
```

Reply:
`(true, true)` should be an error, no? as constant is a contract not to read or write state, both being set would be an invalid ABI?